### PR TITLE
KAFKA-14279; Add 3.3.x streams system tests

### PR DESCRIPTION
**Pull request by jsancio**
Add 3.3.x stream upgrade tests. The streams/upgrade-system-tests-33 directory was created by coping streams/upgrade-system-tests-32 without changes.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jsancio** [comment]
@cadonna can you review this PR?

Running system tests on this branch.

**Comment by jsancio** [comment]
Here are the system test results: http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/system-test-kafka-branch-builder--1672893494--jsancio--kafka-14545-stream-3-3-test--a595bcfb9c/2023-01-04--001./2023-01-04--001./report.html

Looks like this test failed:
```
Module: kafkatest.tests.streams.streams_upgrade_test
Class:  StreamsUpgradeTest
Method: test_rolling_upgrade_with_2_bounces
Arguments:
{
  "from_version": "3.3.1",
  "to_version": "3.5.0-SNAPSHOT"
}
```

I am going to rerun just this test.

File: streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    @SuppressWarnings("unchecked")
+    public static void main(final String[] args) throws Exception {
+        if (args.length < 1) {
+            System.err.println("StreamsUpgradeTest requires one argument (properties-file) but provided none");
+        }
+        final String propFileName = args[0];
+
+        final Properties streamsProperties = Utils.loadProps(propFileName);
+
+        System.out.println("StreamsTest instance started (StreamsUpgradeTest v3.2)");
```
Comment (mjsax) [inline streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]:
```suggestion
        System.out.println("StreamsTest instance started (StreamsUpgradeTest v3.3)");
```

File: streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        kStream.to("fk-result", Produced.with(stringSerde, stringSerde));
+    }
+
+    private static <KIn, VIn, KOut, VOut> ProcessorSupplier<KIn, VIn, KOut, VOut> printProcessorSupplier(final String topic) {
+        return () -> new ContextualProcessor<KIn, VIn, KOut, VOut>() {
+            private int numRecordsProcessed = 0;
+
+            @Override
+            public void init(final ProcessorContext<KOut, VOut> context) {
+                System.out.println("[3.2] initializing processor: topic=" + topic + "taskId=" + context.taskId());
```
Comment (mjsax) [inline streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]:
```suggestion
                System.out.println("[3.3] initializing processor: topic=" + topic + "taskId=" + context.taskId());
```

File: streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    static ProcessorSupplier<Object, Object, Void, Void> printProcessorSupplier(final String topic, final String name) {
+        return () -> new ContextualProcessor<Object, Object, Void, Void>() {
+            private int numRecordsProcessed = 0;
+            private long smallestOffset = Long.MAX_VALUE;
+            private long largestOffset = Long.MIN_VALUE;
+
+            @Override
+            public void init(final ProcessorContext<Void, Void> context) {
+                super.init(context);
+                System.out.println("[3.2] initializing processor: topic=" + topic + " taskId=" + context.taskId());
```
Comment (mjsax) [inline streams/upgrade-system-tests-33/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java]:
```suggestion
                System.out.println("[3.3] initializing processor: topic=" + topic + " taskId=" + context.taskId());
```

**Review by mjsax — COMMENTED** [review]
A few minor comments.

**Comment by jsancio** [comment]
Same failure again: http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/system-test-kafka-branch-builder--1672950817--jsancio--kafka-14545-stream-3-3-test--a595bcfb9c/2023-01-05--001./2023-01-05--001./report.html

`TimeoutError('Could not detect Kafka Streams version 3.5.0-SNAPSHOT on ubuntu@worker29')`

**Comment by mjsax** [comment]
@jsancio -- any update?

**Comment by jsancio** [comment]
@mjsax I got a chance to work on this today. I updated the PR. Here is the result from running the test locally:
```bash
TC_PATHS="tests/kafkatest/tests/streams/streams_upgrade_test.py::StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces" _DUCKTAPE_OPTIONS='--parameters '\''{"from_version":"3.3.1","to_version":"3.5.0-SNAPSHOT"}'\' tests/docker/run_tests.sh

> Configure project :
Starting build with version 3.5.0-SNAPSHOT (commit id a0090d24) using Gradle 7.6, Java 1.8 and Scala 2.13.10
Build properties: maxParallelForks=12, maxScalacThreads=8, maxTestRetries=0

BUILD SUCCESSFUL in 1s
168 actionable tasks: 168 up-to-date
docker exec ducker01 bash -c "cd /opt/kafka-dev && ducktape --cluster-file /opt/kafka-dev/tests/docker/build/cluster.json  ./tests/kafkatest/tests/streams/streams_upgrade_test.py::StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces --parameters '{"from_version":"3.3.1","to_version":"3.5.0-SNAPSHOT"}'"
/usr/local/lib/python3.9/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
[INFO:2023-01-09 16:14:29,112]: starting test run with session id 2023-01-09--003...
[INFO:2023-01-09 16:14:29,112]: running 1 tests...
[INFO:2023-01-09 16:14:29,112]: Triggering test 1 of 1...
[INFO:2023-01-09 16:14:29,117]: RunnerClient: Loading test {'directory': '/opt/kafka-dev/tests/kafkatest/tests/streams', 'file_name': 'streams_upgrade_test.py', 'cls_name': 'StreamsUpgradeTest', 'method_name': 'test_rolling_upgrade_with_2_bounces', 'injected_args': {'from_version': '3.3.1', 'to_version': '3.5.0-SNAPSHOT'}}
[INFO:2023-01-09 16:14:29,118]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: on run 1/1
[INFO:2023-01-09 16:14:29,119]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: Setting up...
[INFO:2023-01-09 16:14:29,119]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: Running...
[INFO:2023-01-09 16:18:09,123]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: Tearing down...
[INFO:2023-01-09 16:18:28,193]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: PASS
[INFO:2023-01-09 16:18:28,193]: RunnerClient: kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT: Data: None
================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.11.1
session_id:       2023-01-09--003
run time:         3 minutes 59.087 seconds
tests run:        1
passed:           1
flaky:            0
failed:           0
ignored:          0
================================================================================
test_id:    kafkatest.tests.streams.streams_upgrade_test.StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces.from_version=3.3.1.to_version=3.5.0-SNAPSHOT
status:     PASS
run time:   3 minutes 59.075 seconds
--------------------------------------------------------------------------------

```

**Comment by mjsax** [comment]
Thanks for the PR. Merged to `trunk` and cherry-picked to `3.4` branch.

**Pull request merged**